File: JAVA (본인)/chap06/src/sec01/exam01/Array.java
package sec01.exam01;

public class Array {

//	int[] a = new int[3];
//	Array() {
//		
//		
//		
//		System.out.println(a.length);

//		a.length  = 5;
	// 길이 변경 불가 (final)
//	}

	final int length;
	// 고정된 수의 변수
//
//	int x;
//	int index;
	int[] arr;
	int cursor = 0;
	// 배열 값 치환

	
	
	Array(int length) {
		this.length = length; 
		arr = new int[length];
		              // 배열 응용
	}

	
	
	
	void add(int x) {

//	this.x = index;
                      // 배열 index가 임의숫자보다 높을수 없음
		if (cursor > this.length) {

			System.out.println("에러 문구 출력");
		}

		arr[cursor] = x;
		cursor++; // 1씩 상승

		// add에 값을 넣음
		// 현재 index에 추가
		// 배열 목차에 넣음
		
		
		
		
		// index 이상이면
		// index > get 메소드의 매개변수.
		// "에러 문구 출력"

	}

	
	
	
	
	
	int get(int index) {

		if (index < this.length) {
			return arr[index];
			

		} else {
			
			
			System.out.println("범위를 초과했습니다");
			
		}

	

//    	  int[] b = new int[3];
//        b[0] = 100;
//        b[1] = 200;
//        
//        System.out.println(b[0]);

		return index;
	}

	
	
	
	
	
	
}

File: html/JS/array.js
// 배열은 변수의 집합
// 한번에 몽땅 만들어서 index로 관리할 수 있는 변수의 집합

// 배열 변수를 만드는 가장 쉬운 방법
let arr = [];
// 두번째 방법
let arr2 = new Array();

// 값을 미리 알고 있을 때
arr = [1,2,3];
arr = new Array(1,2,3);
console.log("arr[0]", arr[0]); // 1


arr[1] = 20;
console.log("arr[1]", arr[1]); // 20

arr[10] = 10;
console.log("arr", arr);
console.log("arr[20]", arr[20]); // undifined
// 자바 객체 : 클래스



arr["name"] = "이름";
console.log("arr", arr);
console.log(`arr["name"]`, arr["name"]); // 이름 (임의의 값)
console.log("arr.length", arr.length)



const str = "휴먼교육센터";
console.log("str[1]", str[1]); // 먼
console.log("str.length", str.length); //6


// arr = [휴, 먼, 교, 육, 센, 터];
for(let i=0; i<=str.length; i++) {
    console.log(str[i], i);
}


// for(let i=0; i<=arr.length; i++) {
//     console.log(arr[i]); 
// }







let 액션 = [`헬다이버2`, `엘든링`, `사펑`, `콜옵`];
let 롤플레잉 = [`발더3`, `호구와트`, `스토커2`];
let 전략 = [`발트로`, `토탈워`, `메타포`, `스텔라리스`];

let 스팀 = [액션, 롤플레잉, 전략];

for(let j=0; j<스팀.length; j++) {

    // console.log(스팀[j]);

    for(let i=0; i<스팀[j].length; i++) {

        console.log(스팀[j][i], j, i);
    }

}

// for(let i=0; i<액션.length; i++) {

//     console.log(액션[i]);
// }




// 배열의 초기화
arr = []; 
arr = [1,3,2,5,4,2];

//js에서 배열을 java의 ArrayList처럼 메소드 사용하기

//마지막에 하나 추가
// 자동으로 배열 크기도 증가
// java의 ArrayList의 add와 같음
arr.push(10);
console.log(arr);

// 맨 처음에 하나 추가
arr.unshift(11);
console.log(arr);

let pop = arr.pop();
console.log(arr);
console.log("pop : ", pop);

let shift = arr.shift();
console.log(arr);
console.log("shift : ", shift);

// 배열 순서 거꾸로
arr.reverse();
console.log(arr);

// 배열 정렬, 단 오름차순만 지원
arr.sort();
console.log(arr);

// [1, 2, 2, 3, 4 ,5]
// splice(시작 index, count)
// splice(시작 index)  
let splice = arr.splice(2, 3);
console.log("arr", arr);
console.log("splice", splice);

// 1, 2, 5
// 음수 index는 뒤에서부터
let splice2 = arr.splice[-2]
console.log("arr", arr);
console.log("splice2", splice2);

// let month = "12";
// let month2 = "1";

// month = "0" + month; // "012"
// month2 = "0"+ month2; //"01"

arr = [1,2,3,4,5];
// 1,2는 똑같이 동작하되 지운 자리에 3번째 전달인자 이후의 값을 넣음
let s = arr.splice(1,2, `둘`)
console.log(arr)
console.log(s)

let join = arr.join()
let join2 = arr.join(`-`)
console.log(arr)
console.log(join)
console.log(join2)

// 문자열 함수 split을 이용해서
// 전달인자의 문자로 잘라서 배열로 만들어 줌
// 단, 전달인자의 문자는 사라짐
let split = join2.split("-")
console.log(split)

arr = [1,2,3,4,5]
//slice (시작 index, 종료index)
// Ctrl + c 느낌
let slice = arr.slice(1,3);
console.log(arr)
console.log(slice)
console.log(concat)

console.log(arr.toString());
console.log(""+ arr);
